### ISSUE TITLE
Date mask 00/00/00 does not handle existing dates with diffrent mask

### ISSUE THREAD

**Issue by dkorobov** [igorescobar/jQuery-Mask-Plugin#536]
Date mask 00/00/00 incorrectly masks preexisting dates with 00/00/0000 mask so instead of 04/26/17 its 04/26/20 given 04/26/2017. I can try to contribute tomorrow if possible.

**Comment by igorescobar** [comment]
@dkorobov I can't help you if you don't provide a functional jsfiddle :(

**Comment by igorescobar** [comment]
@dkorobov but I can say for sure that this is the expected behaviour. This plugin masks what the user types into something readable. The information came from your server in a different format and what the plugin did was try to cut the leftovers and remains the structure. 

What you can do here is you can pass a function instead of a string in place of the mask argument, like this:

```javascript
const el = $('.sp_celphones');
const maskAsFunction = (val) => {
  const dateMask = /[0-9]{2}\/[0-9]{2}\/[0-9]{4}/;
  
  if (val.match(dateMask)) {
    el.val('.....'); // convert the string to whatever you what
  }

  return '00/00/0000';
};

el.mask(maskAsFunction);
```

**Comment by dkorobov** [comment]
Yeah I did that once then twice and I'm thinking I don't want to spread the same snippet everywhere. I'm in a situation where I have no control over the server code. The server keeps sending 4 digit year and users insist on filling out 2 digit years. I'll try to branch today and add an option for mask conversion for the mm/dd/yyyy to mm/dd/yy case.

**Comment by igorescobar** [comment]
@dkorobov you dont need to this everywhere. just define a class ".date" and every place that has the same class will inherit the same behaviour. jQuery mask has a generic algorithm it doesn't know what is a date or what is not. He just do what is told without know any context.

**Comment by dkorobov** [comment]
Yeah I have no way to do that cause I don't own the system. I work with documents and they come prestructured. I'll try to add a context option so the value can be read knowing the server mask. I'll post where I get with this. Thanks for your help. Didn't expect fast response.

**Comment by dkorobov** [comment]
I looked through the source code and i would say ideally i would use a mask '00/00/xx00' but that would make it into a rather irregular expression and changing values inside the plugin based on starting mask is more work and is probably not very functional from the plugins stand point. That's more of an input issue rather then the plugins function. I was able to get the system owner to host my function at the master level. I ended up with basically the same function use suggested.
